api/capi/src/nntrainer.cpp

```diff
@@ -23,8 +23,8 @@
 #include <databuffer.h>
 #include <databuffer_file.h>
 #include <databuffer_func.h>
-#include <ml-api-common.h>
```

Comment:
huh.. why remove this? ml_train_dataset_create_with_generator() needs this.

Reply:
I think '*api' can and should solely depend on the <nntrainer_error.h> since #307.
<nntrainer_error.h> is superset of <ml-api-common.h> we have to make a boundary for that.

(theoretically) declaring error code twice otherwise.

Reply:
aah, I confused with nntrainer-api-common.h